From ed0b28e9a115f8353a5c533c3d55244590024df2 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Tue, 24 Jan 2017 21:28:14 +0200
Subject: [PATCH 0763/1345] fix: dts: a80x0: a70x0: move l4_chksum_jumbo_port
 to board level dts

commit  82e169be9ca0799523b3674a8fa7de538fafb84a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Issue:
- Warnings due to disabled port 0 in board level dts and port 0
  selected to support jumbo L4 checksum in armada-cp110.dtsi

- l4_chksum_jumbo_port set to port 0 in SoC level dts for all
  board level configurations. In some board level dts files port 0
  disabled and in this case disabled port set to support jumbo L4
  checksum.

- To prevent this issue l4_chksum_jumbo_port moved to board
  level dts.

Fix:
- Move l4_chksum_jumbo_port from cp110 architecture dts to
  board level dts

- For armada-7040-db-all-pcie and armada-8040-db-B
  configurations with disabled port 0, jumbo L4 checksum support set
  for port 1 in armada-7040-db-all-pcie and for port 2 in
  armada-8040-db-B on CP1 in other board level dts files port 0 set to
  support jumbo L4 checksum.

Change-Id: I15d552e25f2a278a14ab139fcd6ee41b75bfdd11
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/36005
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-A.dts   |    1 +
 .../boot/dts/marvell/armada-7040-db-all-pcie.dts   |    1 +
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |    1 +
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts   |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts  |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-db-B.dts   |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-db-C.dts   |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-db-D.dts   |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-db-E.dts   |    2 ++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |    2 ++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      |    1 -
 11 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
index 3822741..62063f4 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-A.dts
@@ -151,6 +151,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				eth0: eth0@010000 {
 					status = "okay";
 				};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
index de7fd70..f1db8b0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
@@ -138,6 +138,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <1>;
 				eth0: eth0@010000 {
 					status = "disabled";
 				};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index ea95f23..8aee146 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -149,6 +149,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				eth0: eth0@010000 {
 					status = "okay";
 				};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index a888b54..86d2f5b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -127,6 +127,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
 
@@ -198,6 +199,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_ge1_rgmii_pins>;
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
index 22b192a..d0b504f 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A0.dts
@@ -130,6 +130,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
 
@@ -193,6 +194,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_ge1_rgmii_pins>;
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
index acdf5f2..a2389af 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-B.dts
@@ -112,6 +112,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <2>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
 
@@ -179,6 +180,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_ge1_rgmii_pins>;
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
index 0cc83d1..77bf863 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-C.dts
@@ -127,6 +127,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
 
@@ -190,6 +191,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				eth0_1: eth0@010000 {
 					status = "okay";
 				};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
index 67c5fe7..bb9381a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
@@ -127,6 +127,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cpm_ge2_rgmii_pins>;
 
@@ -190,6 +191,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_ge1_rgmii_pins>;
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
index a5babc2..8c0c18c 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-E.dts
@@ -131,6 +131,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				eth0: eth0@010000 {
 					status = "okay";
 				};
@@ -199,6 +200,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				pinctrl-0 = <&cps_ge1_rgmii_pins>;
 
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 656f3e6..b29972a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -90,6 +90,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				eth0_0: eth0@010000 {
 					status = "okay";
@@ -130,6 +131,7 @@
 				};
 			};
 			ppv22@000000 {
+				l4_chksum_jumbo_port = <0>;
 				pinctrl-names = "default";
 				eth0_1: eth0@010000 {
 					status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 38c4dab..dbc4ee6 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -395,7 +395,6 @@ ppv22@000000 {
 		 <&cpm_syscon0 1 6>, <&cpm_syscon0 1 5>;
 	clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
 	status = "okay";
-	l4_chksum_jumbo_port = <0>;
 	eth0: eth0@010000 {
 		interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
 			     <ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-- 
1.7.9.5

